### ISSUE TITLE
Prefer Spread should allow apolloLink concat method

### ISSUE THREAD

**Issue by Nibblesh** [sindresorhus/eslint-plugin-unicorn#1106]
When using ApolloLink, there is a method `concat` that you can use to join multiple links

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-spread.md

the rule fails the ApolloLink concat method without checking if it is an iterable

https://www.apollographql.com/docs/link/composition/#additive-composition

```js
import { ApolloLink } from 'apollo-link';
import { RetryLink } from 'apollo-link-retry';
import { HttpLink } from 'apollo-link-http';

const link = ApolloLink.concat(new RetryLink(), new HttpLink({ uri: 'http://localhost:4000/graphql' }));
```

**Comment by sindresorhus** [comment]
JavaScript doesn't statically know any types in most cases, so this is impossible. The only solution here would be to use the TypeScript parser and use the types it manages to infer: https://github.com/sindresorhus/eslint-plugin-unicorn/issues/347

However, we could potentially hard-code an ignore for `ApolloLink.concat` specifically.

**Comment by fisker** [comment]
This is already fixed in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1069, any object starts with a uppercase letter tread as static method of `Buffer.concat`/ `ApolloLink.concat` same.

Are you using an old version of `eslint-plugin-unicorn`?

Closing this issue, but feel free to continue discussion if necessary.
